### ISSUE TITLE
core dump when use official example trtllm-build decoder

### ISSUE THREAD

**Issue by XuechenCao** [NVIDIA/TensorRT-LLM#2886]
### System Info

nvidia-smi
Thu Mar 13 16:59:31 2025       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 570.86.15              Driver Version: 570.86.15      CUDA Version: 12.8     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 4070        Off |   00000000:01:00.0 Off |                  N/A |
|  0%   38C    P8             10W /  200W |      18MiB /  12282MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
                                                                                         
+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A            2735      G   /usr/lib/xorg/Xorg                        4MiB |
+-----------------------------------------------------------------------------------------+

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

after success build encoder, try to build decoder with core dump
trtllm-build  --checkpoint_dir ${checkpoint_dir}/decoder \
              --output_dir ${output_dir}/decoder \
              --moe_plugin disable \
              --max_beam_width ${MAX_BEAM_WIDTH} \
              --max_batch_size ${MAX_BATCH_SIZE} \
              --max_seq_len 114 \
              --max_input_len 14 \
              --max_encoder_input_len 3000 \
              --gemm_plugin ${INFERENCE_PRECISION} \
              --bert_attention_plugin ${INFERENCE_PRECISION} \
              --gpt_attention_plugin ${INFERENCE_PRECISION}
[TensorRT-LLM] TensorRT-LLM version: 0.17.0.dev2024121700
[03/13/2025-17:09:54] [TRT-LLM] [I] Set bert_attention_plugin to float16.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set gpt_attention_plugin to float16.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set gemm_plugin to float16.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set gemm_swiglu_plugin to None.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set fp8_rowwise_gemm_plugin to None.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set nccl_plugin to auto.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set lora_plugin to None.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set moe_plugin to None.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set mamba_conv1d_plugin to auto.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set low_latency_gemm_plugin to None.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set low_latency_gemm_swiglu_plugin to None.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set context_fmha to True.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set bert_context_fmha_fp32_acc to False.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set remove_input_padding to True.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set reduce_fusion to False.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set user_buffer to False.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set tokens_per_block to 64.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set use_paged_context_fmha to False.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set use_fp8_context_fmha to False.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set multiple_profiles to False.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set paged_state to True.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set streamingllm to False.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set use_fused_mlp to True.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set pp_reduce_scatter to False.
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.use_prompt_tuning = False
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.has_position_embedding = True
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.layernorm_type = 0
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.has_attention_qkvo_bias = True
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.has_mlp_bias = True
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.has_model_final_layernorm = True
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.has_embedding_layernorm = False
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.has_embedding_scale = False
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.ffn_hidden_size = 5120
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.q_scaling = 1.0
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.layernorm_position = 0
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.relative_attention = False
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.max_distance = 0
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.num_buckets = 0
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.model_type = whisper
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.rescale_before_lm_head = False
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.encoder_hidden_size = 1280
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.encoder_num_heads = 20
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.encoder_head_size = None
[03/13/2025-17:09:54] [TRT-LLM] [W] Implicitly setting PretrainedConfig.skip_cross_kv = False
[03/13/2025-17:09:54] [TRT-LLM] [I] Compute capability: (8, 9)
[03/13/2025-17:09:54] [TRT-LLM] [I] SM count: 46
[03/13/2025-17:09:54] [TRT-LLM] [I] SM clock: 3105 MHz
[03/13/2025-17:09:54] [TRT-LLM] [I] int4 TFLOPS: 292
[03/13/2025-17:09:54] [TRT-LLM] [I] int8 TFLOPS: 146
[03/13/2025-17:09:54] [TRT-LLM] [I] fp8 TFLOPS: 146
[03/13/2025-17:09:54] [TRT-LLM] [I] float16 TFLOPS: 73
[03/13/2025-17:09:54] [TRT-LLM] [I] bfloat16 TFLOPS: 73
[03/13/2025-17:09:54] [TRT-LLM] [I] float32 TFLOPS: 36
[03/13/2025-17:09:54] [TRT-LLM] [I] Total Memory: 11 GiB
[03/13/2025-17:09:54] [TRT-LLM] [I] Memory clock: 10001 MHz
[03/13/2025-17:09:54] [TRT-LLM] [I] Memory bus width: 192
[03/13/2025-17:09:54] [TRT-LLM] [I] Memory bandwidth: 480 GB/s
[03/13/2025-17:09:54] [TRT-LLM] [I] PCIe speed: 2500 Mbps
[03/13/2025-17:09:54] [TRT-LLM] [I] PCIe link width: 16
[03/13/2025-17:09:54] [TRT-LLM] [I] PCIe bandwidth: 5 GB/s
[03/13/2025-17:09:54] [TRT-LLM] [I] Set dtype to float16.
[03/13/2025-17:09:54] [TRT-LLM] [I] Set paged_kv_cache to True.
[03/13/2025-17:09:54] [TRT-LLM] [W] Overriding paged_state to False
[03/13/2025-17:09:54] [TRT-LLM] [I] Set paged_state to False.
[03/13/2025-17:09:54] [TRT-LLM] [W] remove_input_padding is enabled, while opt_num_tokens is not set, setting to max_batch_size*max_beam_width. 

[03/13/2025-17:09:54] [TRT-LLM] [W] max_num_tokens (912) shouldn't be greater than max_seq_len * max_batch_size (912), specifying to max_seq_len * max_batch_size (912).
[03/13/2025-17:09:54] [TRT-LLM] [W] padding removal and fMHA are both enabled, max_input_len is not required and will be ignored
[03/13/2025-17:09:55] [TRT] [I] [MemUsageChange] Init CUDA: CPU +3, GPU +0, now: CPU 175, GPU 174 (MiB)
[03/13/2025-17:09:56] [TRT] [I] [MemUsageChange] Init builder kernel library: CPU +2283, GPU +440, now: CPU 2613, GPU 614 (MiB)
[03/13/2025-17:09:56] [TRT-LLM] [I] Set weight_only_quant_matmul_plugin to float16.
[03/13/2025-17:09:56] [TRT-LLM] [I] Set nccl_plugin to None.
[03/13/2025-17:09:56] [TRT-LLM] [I] Total time of constructing network from module object 1.3754022121429443 seconds
[03/13/2025-17:09:56] [TRT-LLM] [I] Total optimization profiles added: 1
[03/13/2025-17:09:56] [TRT-LLM] [I] Total time to initialize the weights in network Unnamed Network 0: 00:00:00
[03/13/2025-17:09:56] [TRT-LLM] [I] Build TensorRT engine Unnamed Network 0
[03/13/2025-17:09:56] [TRT] [W] Unused Input: host_kv_cache_block_offsets
[03/13/2025-17:09:56] [TRT] [W] Unused Input: cross_kv_cache_gen
[03/13/2025-17:09:56] [TRT] [W] [RemoveDeadLayers] Input Tensor host_kv_cache_block_offsets is unused or used only at compile-time, but is not being removed.
[03/13/2025-17:09:56] [TRT] [W] [RemoveDeadLayers] Input Tensor cross_kv_cache_gen is unused or used only at compile-time, but is not being removed.
[03/13/2025-17:09:56] [TRT] [I] Global timing cache in use. Profiling results in this builder pass will be stored.
[03/13/2025-17:09:56] [TRT] [I] Compiler backend is used during engine build.
[cxc-System-Product-Name:192227] *** Process received signal ***
[cxc-System-Product-Name:192227] Signal: Aborted (6)
[cxc-System-Product-Name:192227] Signal code:  (-6)
Aborted (core dumped)

### Expected behavior

output the config and engine without error

### actual behavior

coredump

### additional notes

none

**Comment by dominicshanshan** [comment]
@XuechenCao could you try to use new main branch to see if happen again?

**Comment by karljang** [comment]
Closing as stale since the TensorRT engine-build path is deprecated in favor of the PyTorch backend. Please open a new issue if the problem persists in the latest version.
